File: routerapp/lib/models/rota.dart
class Rota {
  final int id;
  final String codigo;
  final DateTime datacriacao;
  final DateTime? datafechamento;
  final String? observacao;
  final int ativo;
  final int idsituacao;

  Rota({
    required this.id,
    required this.codigo,
    required this.datacriacao,
    this.datafechamento,
    this.observacao,
    required this.ativo,
    required this.idsituacao,
  });

  factory Rota.fromJson(Map<String, dynamic> json) {
    return Rota(
      id: json['id'],
      codigo: json['codigo'],
      datacriacao: DateTime.parse(json['datacriacao']),
      datafechamento: json['datafechamento'] != null
          ? DateTime.tryParse(json['datafechamento'])
          : null,
      observacao: json['observacao'],
      ativo: json['ativo'],
      idsituacao: json['idsituacao'],
    );
  }
}

File: WebApiRouter/Models/Rota.cs
using System;
using System.Collections.Generic;

namespace WebApiRouter.Models;

/// <summary>
/// CADASTRO DAS ROTAS
/// </summary>
public partial class Rota
{
    public int Id { get; set; }

    public string Codigo { get; set; } = null!;

    public DateTime Datacriacao { get; set; }

    public DateTime? Datafechamento { get; set; }

    public string? Observacao { get; set; }

    /// <summary>
    /// COLUNA PARA VALIDAR SE A ROTA ESTÁ ATIVA OU NÃO.
    /// </summary>
    public sbyte Ativo { get; set; }

    public int SituacaoRotaId { get; set; }

    public virtual ICollection<Servico> Servicos { get; set; } = new List<Servico>();

    public virtual SituacaoRota SituacaoRota { get; set; } = null!;
}
